src/pylibsshext/session.pyx

```diff
@@ -243,6 +243,16 @@ cdef class Session(object):
             if (key in OPTS_MAP or key in OPTS_DIR_MAP) and (kwargs[key] is not None):
                 self.set_ssh_options(key, kwargs[key])
 
+        if 'config_file' in kwargs:
```

Comment:
This new key needs to be documented in the doc text above with the description of how the corner cases are handled.

The libssh itself reads the OpenSSH configuration files by default and the openssh configuration files have awkward precedence, so this documentation needs to clarify if this configuration file is parsed before the openssh ones, after them and if the same option is specified in both config and direct option, which one is effective.

@donnerhacke Please, let me know if you plan to get back to make this working or you abandoned this PR in favor of different solution (such as using the OpenSSH config directly).